wordpress-mobile/WordPress-Android
# Implement my domains screen

Fixes #19072 

### Description

This PR implements the "My Domains" screen, with an entry point on the "Me" tab of the main activity.

To test:

Pre-requisites: Please test this screen with an account with active domains.

1. Enable the Domain Management feature flag (in debug settings)
2. Open the My domains screen: Me -> Domains -> My Domains

The initial "ghost" screen should load while the domains list is fetched. It should then display the list of domains as pictured below (in various modes):

| State | Light | Dark | RTL |
|-|-|-|-|
| `Initial` | ![image](https://github.com/wordpress-mobile/WordPress-Android/assets/8507675/c523f468-acc9-4907-abca-ce8c8d6059b2) | ![image](https://github.com/wordpress-mobile/WordPress-Android/assets/8507675/29e89565-7c16-4471-95fd-dd1a0959a1e6) | ![image](https://github.com/wordpress-mobile/WordPress-Android/assets/8507675/03d6275f-2cc1-4cc1-9a7d-4868c72fcc55) |
| `Loaded` | ![my-domains-light](https://github.com/wordpress-mobile/WordPress-Android/assets/8507675/780f12a3-5b11-452b-8aad-b1d482653200) | ![my-domains-dark](https://github.com/wordpress-mobile/WordPress-Android/assets/8507675/c8b681d6-ea6a-40d2-beb8-a51fbb0f57cb) | ![image](https://github.com/wordpress-mobile/WordPress-Android/assets/8507675/5ac3957c-573f-4c75-9ab9-0d4eb6c347dd) |


## Regression Notes
1. Potential unintended areas of impact
N/A

4. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

5. What automated tests I added (or what prevented me from doing so)
Unit tests will be added later, as the view model logic is wired up to other flows.

PR submission checklist:


UI Changes testing checklist:
